HealthMonitoring/Model/EndpointMetadata.cs

```diff
@@ -1,18 +1,33 @@
+using System;
 using System.Linq;
 
 namespace HealthMonitoring.Model
 {
     public class EndpointMetadata
     {
-        public EndpointMetadata(string name, string group, string[] tags)
+        public EndpointMetadata(string name, string group, string[] tags, DateTime? firstTimeRegistered = null, DateTime? lastTimeRegistrationUpdated = null)
```

Comment:
When those times could be null? If this is for migration, then the db migration should be maybe updated to provide values for existing records